### PR TITLE
[Python] Add from_parquet features

### PR DESCRIPTION
**Pull request by papparapa**
[#5323](https://github.com/duckdb/duckdb/issues/5323)
Added support for various `parquet_scan` variables in `from_parquet`.
Now it appcepts a list of `str`s to specify parquet files, like

```Python
duckdb.from_parquet(['t1.parquet','t2.parquet'])
```

It also accepts a `file_row_number`, `filename` and `hive_partitioning` as keyword args, like
```Python
duckdb.from_parquet('t1.parquet', file_row_number=True, filename=True)
```

### CODE REVIEW

File: tools/pythonpkg/duckdb_python.cpp [hunk 1]
```diff
@@ -115,6 +115,14 @@ PYBIND11_MODULE(DUCKDB_PYTHON_LIB_NAME, m) {
 	m.def("from_parquet", &DuckDBPyRelation::FromParquet,
 	      "Creates a relation object from the Parquet file in file_name", py::arg("file_name"),
 	      py::arg("binary_as_string"), py::arg("connection") = py::none());
+	m.def("from_parquet", &DuckDBPyRelation::FromParquetGlob,
```
Comment (Mause) [inline tools/pythonpkg/duckdb_python.cpp:118]: Can you please collapse these three methods into one, with optional arguments? Python doesn't handle function overloads very well

Comment (papparapa) [reply tools/pythonpkg/duckdb_python.cpp:118]: You mean pushing the overload to C++ instead of Python, right?

Comment (Mause) [reply tools/pythonpkg/duckdb_python.cpp:118]: More or less yes, it should be a single method that has optional extra keyword arguments

Comment (papparapa) [reply tools/pythonpkg/duckdb_python.cpp:118]: OK. I'll try

**Comment by papparapa** [comment]
Collapsed four overloads into two, one takes a `str` and the other takes a list of `str`s.

**Comment by Mause** [comment]
You'll also need to update the python pyi stubs, using the [`scripts/regenerate_python_stubs.sh`] script

**Comment by papparapa** [comment]
I had missed the documentation.
Thanks!

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM

**Pull request merged**